src/main/kotlin/day16/data_source/ImageDataSourceImpl.kt

```
@@ -0,0 +1,35 @@
+import io.ktor.client.*
+import io.ktor.client.engine.cio.*
+import io.ktor.client.request.*
+import io.ktor.client.statement.*
+import kotlinx.io.IOException
+import java.io.File
+
+class ImageDataSourceImpl(private val client: HttpClient = HttpClient(CIO)) : ImageDataSource {
+    override suspend fun fetchImage(url: String): ByteArray {
+        return try {
+            client.get(url).bodyAsBytes()
+        } catch (e: Exception) {
```

detekt (1.23.7)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)